packages/backend-common/src/database/types.ts

```diff
@@ -30,6 +30,21 @@ export interface PluginDatabaseManager {
    * stores so that plugins are discouraged from database integration.
    */
   getClient(): Promise<Knex>;
+
+  /**
+   * This property is used to control the behavior of database migrations.
+   */
+  migrations: {
+    /**
+     * apply can be used to determine if database migrations
+     * should be performed.
+     *
+     * Useful if connecting to a read-only database.
+     *
+     * @default true
+     */
+    apply: boolean;
```

Comment:
@marcuseide I think it would be nicer to rename this to `skip` instead. That would make it default to false so users wouldn't have to look in docs to realise that apply is true by default.

Secondly does it make sense to make both this field and migrations optional?